imagedephi/redact/svs.py

```diff
@@ -151,3 +151,35 @@ def execute_plan(self) -> None:
                 self.apply(rule, ifd)
             elif tag.value == image_description_tag.value:
                 self._redact_svs_image_description(ifd)
+
+
+class SvsImageRedactionPlan(TiffImageRedactionPlan):
+    file_format = FileFormat.SVS
+
+    def get_associated_image_key_for_ifd(self, ifd: IFD):
+        """Attempt to match an associated image with a rule. Fallback to default rule."""
+        image_description = str(ifd["tags"][270]["data"])
+        # we could do additional checks, like look for a macro based on dimensions
+        for key in self.final_rules:
+            if key in image_description:
+                return key
+        return "default"
```

Comment:
@manthey is there a better way to do determine if an IFD in an svs file is a macro or label? I remember you saying that macro images are usually much wider than they are tall, due to the dimensions of the slides themselves.

I also remember you saying that if svs files don't have a macro, they won't work properly with Aperio software, so I want to make sure we properly identify macro images as often as possible.

Reply:
Thanks for the correction. As part of this PR I've added new base rules, including replace-with-blank-image for label images in svs files.